scripts/core/superdesk-api.d.ts

```diff
@@ -963,6 +963,8 @@ declare module 'superdesk-api' {
             formatFiltersForServer?: (filters: ICrudManagerFilters) => ICrudManagerFilters,
         ): Promise<IRestApiResponse<T>>;
         queryRaw<T>(endpoint, params?: Dictionary<string, any>): Promise<T>;
+        queryZip<T>(endpoint, params?: Dictionary<string, any>): Promise<void>;
+        queryAssetBinary<T>(endpoint): any;
```

Comment:
This function doesn't return anything, so `void` would be better than `any`

Reply:
What do these new methods do?

Reply:
queryZip is used to request compressed binary zip for multiple assets and download the returned compressed binary as zip. 
And queryAssetBinary is used to open a single asset into a new window.
Can not use queryRaw as response from server is Binary and not Json.

Reply:
I see. It should not be in this part of the API though. If I'm not mistaken `queryZip` does 2 things - queries using `httpRequestBase` to receive a blob and downloads that blob to a browser.

I suggest to rename `queryRaw` to `queryRawJson` and create `queryRaw` that would be calling `httpRequestBase` in order to return the response that hasn't been converted to JSON.

After that, you could add a method to helpers or utilities selection `download(blob, applicationType): void` that would do the downloading separately.

`queryAssetBinary` doesn't query anything and doesn't even initiate an AJAX request. I would drop it entirely since it's basically a wrapper for `window.open`. You can get base server url any time from config that is [exposed to extensions](https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/core/superdesk-api.d.ts#L996).

Reply:
I agree with you. I'll have a discussion with Mark about the changes tomorrow and update you accordingly.